pkg/abstractions/common/instance.go

```diff
@@ -218,6 +222,12 @@ func (i *AutoscaledInstance) Monitor() error {
 }
 
 func (i *AutoscaledInstance) HandleScalingEvent(desiredContainers int) error {
+	_, span := common.TraceFunc(i.Ctx, "pkg/abstractions/common", "AutoscaledInstance.HandleScalingEvent", i.AppConfig.DebugMode,
+		attribute.String("stub.id", i.Stub.ExternalId))
+	if span != nil {
+		defer span.End()
```

Comment:
One way to clean this up a little bit is to embed the span in another struct with its own `End()` method that checks if the span is nil. I like doing it that way because you avoid having to do the conditional check all over the place.

Reply:
Made this change, not sure it's much better since it keeps the struct in memory for the length of the call, but it's fine for now.